### PR TITLE
Issue #3361: Add preview access rules

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/3361
Replaces https://github.com/backdrop/backdrop/pull/2443

### CODE REVIEW

File: core/modules/node/node.module [hunk 1]
```diff
@@ -1884,18 +1885,19 @@ function node_menu() {
...
  *   TRUE if the user has access to the preview, FALSE otherwise.
  *
  * @see node_menu()
  *
- * @since 1.11.0
+ * @since 1.12.0
  */
-function node_access_preview(Node $node) {
-  if (node_access('create', $node) || node_access('update', $node)) {
+function node_preview_access($node_tempstore_id) {
```
Comment (herbdool) [inline core/modules/node/node.module:1898]: Does this capture when someone previews *before* saving? I thought it wouldn't be able to find a node object in that situation.

Comment (docwilmot) [reply core/modules/node/node.module:1898]: Yes the preview button takes a node object, gives it an ID, stores it in tempstore, then thats loaded on the preview page. Thats the same node object the node edit page creates before `$node->save()`. No NID yet, but a Node nonetheless.

Comment (docwilmot) [reply core/modules/node/node.module:1898]: Should also say tests pass, at least locally. Somethings preventing the sandbox, a merge conflict I think. And there are new tests for just what you ask.

Comment (herbdool) [reply core/modules/node/node.module:1898]: Yes, probably a conflict. Happened to me the other day.

**Comment by docwilmot** [comment]
It seems when I fix merge conflicts using the GitHub Desktop on Windows, strange things happen. And I dont quite know what it means. I just tried to update this branch to 1.x latest code, and got a conflict, fixed it. But now the sandbox is showing "Failed to get code from github" above, with an error about unmerged files, and the test sandboxes are 404. @herbdool please advise if this is mergeable or if something is missing/broken.

**Comment by herbdool** [comment]
Let's see if toggling the switch will fix

**Comment by quicksketch** [comment]
Merged into 1.x. We have a merge conflict on 1.11.x, otherwise I'd merge it there as well. With 1.12 around the corner I don't think we need to trouble with porting.

**Pull request merged**